subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/ComponentSelectorSerializer.java

```diff
@@ -39,12 +39,17 @@
 import org.gradle.internal.serialize.Encoder;
 import org.gradle.util.Path;
 
+import javax.annotation.concurrent.NotThreadSafe;
 import java.io.IOException;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
+/**
+ * A serializer for {@link ComponentSelector} that is not thread-safe and reusable.
```

Comment:
This wording is confusing. Does the `not` apply to reusable? IMO, either of the two wordings would be more clear depending on the intention of this documentation:
1. `that is not thread-safe and not reusable`
2. `that is reusable and not thread-safe`

Reply:
Indeed, corrected to be fully explicit.